trino/trino.go

```diff
@@ -1943,3 +2039,19 @@ func (s *NullSlice3Map) Scan(value interface{}) error {
 	s.Valid = true
 	return nil
 }
+
+type QueryProgressCallbackResponse struct {
+	ID         string
+	QueryStats stmtStats
+}
+
+type queryProgressCallbackPeriod struct {
+	Period           time.Duration
+	LastCallbackTime time.Time
+	LastQueryState   string
+}
+
+type ProgressUpdater interface {
```

Comment:
maybe `QueryProgressCallback`?

Reply:
`ProgressUpdater` is idiomatic: https://go.dev/doc/effective_go#interface-names

Reply:
Makes sense.

Should we maybe then call fields in `Conn`
```
	progressUpdater        ProgressUpdater
	progressUpdaterPeriod  queryProgressCallbackPeriod
```
instead:
```
	callback        ProgressUpdater
	callbackPeriod  queryProgressCallbackPeriod
```

Reply:
changed field names :)